src/components/CreateForm/BasicSectionInfo/BasicSectionInfo.tsx

```diff
@@ -0,0 +1,128 @@
+import type { Control, FieldErrors} from "react-hook-form";
+import { type CompanyFormType } from "../../../schema/company.schema";
+import { Country } from "../../../constants/Country.enum";
+import FormGroup from "../FormGroup/FormGroup";
+import ValidationError from "../ValidationError/ValidationError";
+import InputField from "../InputField/InputField";
+import SelectField from "../SelectField/SelectField";
+import styles from "../CreateForm/CreateForm.module.css";
+import type { ICompany } from "../../../interfaces/Company.interface";
+
+interface BasicSectionInfoProps {
+  control: Control<ICompany, unknown, CompanyFormType>;
+  errors: FieldErrors<CompanyFormType>;
+}
+
+export default function BasicSectionInfo({
+  control,
+  errors,
+}: BasicSectionInfoProps) {
+  const countryOptions = Object.values(Country);
+
+  return (
+    <>
+      <div className={styles.form_group}>
+        <FormGroup
+          htmlFor="name"
+          label="Company name"
+          controllerProps={{
+            name: "name",
+            control,
+            render: ({ field }) => (
+              <InputField
+                type="text"
+                id="name"
+                placeholder="Type company name"
+                {...field}
+                value={typeof field.value === "string" ? field.value : ""}
+              />
+            ),
+          }}
+        />
+        <ValidationError
+          isError={Boolean(errors.name)}
+          message={errors.name?.message}
+        />
+      </div>
```

Comment:
I suggest to make separate component and pass to him props as filed name and others